cli/cmd/generate_gcp.go

```diff
@@ -0,0 +1,780 @@
+			Opts:     []survey.AskOpt{survey.WithValidator(validateGcpRegion)},
+			Response: &config.BucketRegion,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionGcpBucketLocation, Default: config.BucketLocation},
+			Checks:   []*bool{&config.AuditLog, &newBucket, &extraState.ConfigureNewBucketSettings},
+			Response: &config.BucketLocation,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionGcpBucketRetention, Default: "0"},
+			Checks:   []*bool{&config.AuditLog, &newBucket, &extraState.ConfigureNewBucketSettings},
+			Response: &config.LogBucketRetentionDays,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionGcpBucketLifecycle, Default: "-1"},
+			Checks:   []*bool{&config.AuditLog, &newBucket, &extraState.ConfigureNewBucketSettings},
+			Response: &config.LogBucketLifecycleRuleAge,
+		},
+		{
+			Prompt:   &survey.Confirm{Message: QuestionGcpEnableUBLA, Default: config.EnableUBLA},
+			Checks:   []*bool{&config.AuditLog, &newBucket, &extraState.ConfigureNewBucketSettings},
+			Required: true,
+			Response: &config.EnableUBLA,
+		},
+		{
+			Prompt:   &survey.Confirm{Message: QuestionGcpEnableBucketForceDestroy, Default: config.EnableForceDestroyBucket},
+			Checks:   []*bool{&config.AuditLog, &newBucket, &extraState.UseExistingBucket},
+			Required: true,
+			Response: &config.EnableForceDestroyBucket,
+		},
+		{
+			Prompt:   &survey.Confirm{Message: QuestionGcpUseExistingSink, Default: extraState.UseExistingSink},
+			Checks:   []*bool{&config.AuditLog, &newBucket, &extraState.ConfigureNewBucketSettings},
+			Required: true,
+			Response: &extraState.UseExistingSink,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionGcpExistingSinkName, Default: config.ExistingLogSinkName},
+			Checks:   []*bool{&config.AuditLog, &newBucket, &extraState.ConfigureNewBucketSettings, &extraState.UseExistingSink},
+			Required: true,
+			Response: &config.ExistingLogSinkName,
+		},
+	}, config.AuditLog)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func promptGcpExistingServiceAccountQuestions(config *gcp.GenerateGcpTfConfigurationArgs) error {
+	// ensure struct is initialized
+	if config.ExistingServiceAccount == nil {
+		config.ExistingServiceAccount = &gcp.ExistingServiceAccountDetails{}
+	}
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionExistingServiceAccountName, Default: config.ExistingServiceAccount.Name},
+			Response: &config.ExistingServiceAccount.Name,
+			Opts:     []survey.AskOpt{survey.WithValidator(survey.Required)},
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionExistingServiceAccountPrivateKey, Default: config.ExistingServiceAccount.PrivateKey},
+			Response: &config.ExistingServiceAccount.PrivateKey,
+			Opts:     []survey.AskOpt{survey.WithValidator(survey.Required), survey.WithValidator(validateStringIsBase64)},
+		}}); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
We do not need the whole if statement above, you can just execute the function and return the err variable. It will be nil or not, no need to check for nil.